File: Assets/_scripts/fish/behaviour/FishLookAheadBehaviour.js
class FishLookAheadBehaviour extends FishBehaviour{
    
    private var orientationMatcher : OrientationMatching;
    
    function FishLookAheadBehaviour(){
        priority = 1;
    }
    
        
    function Start(){
        orientationMatcher = gameObject.AddComponent(OrientationMatching);
    }
    
    function SelfDestroy(){
        orientationMatcher.SelfDestroy();
        super.SelfDestroy();
    }
    
    function GetSteering() : SteeringOutput{
        if(!orientationMatcher)
            return SteeringOutput.empty;
        
        if(!Utils.Approximately(0, rigidbody.velocity.magnitude))
            orientationMatcher.orientation = rigidbody.velocity;
        else
            orientationMatcher.orientation = transform.forward;
            
        return orientationMatcher.GetSteering();
    }
}

File: Assets/_scripts/fish/behaviour/FishLookAheadBehaviour.cs
using UnityEngine;
using System.Collections;

public class FishLookAheadBehaviour : FishBehaviour
{
    private OrientationMatching orientationMatcher;

    void Start(){
        orientationMatcher = (OrientationMatching)gameObject.AddComponent(typeof(OrientationMatching));
    }

    public override void SelfDestroy(){
        orientationMatcher.SelfDestroy();
        base.SelfDestroy();
    }

    public override SteeringOutput GetSteering(){
        if(!orientationMatcher)
            return SteeringOutput.empty;

        if(!Utils.Approximately(0, rigidbody.velocity.magnitude))
            orientationMatcher.orientation = rigidbody.velocity;
        else
            orientationMatcher.orientation = transform.forward;

        return orientationMatcher.GetSteering();
    }
}